### PR TITLE
WIP: Feature/389 webpack

### LINKED ISSUE #389: Webpack
Because of the time suck that was #341, I'd like to try and make the switch to Webpack. 
I'm throwing this out as an example, since I've already put a ton of time into this Webpack config (modeling it after wd_s Gulp workflow): https://github.com/gregrickaby/MoaningMyrtle. 

**Couple things**

- Our current Gulp workflow to be the model we work towards! Meaning, I'd like our Webpack workflow to be as close to a 1:1 match as possible.

- I'd also like to keep Gulp in a feature branch for several months, to give folks time to transition. We did the same thing when we moved from Grunt --> Gulp.

**Other considerations**

- Drop sass-lint for [stylelint](https://stylelint.io/)
- Update to Babel 7
- Drop Sass Docs
- I want linting to happen with each compile (see my example Webpack config, it does this)
- BrowserSync stays!
- NPM scripts all they way
- Move browerlist to `package.json` 
- Update our linting configs to match WDS Code Standards

### PR DESCRIPTION
**Pull request by gregrickaby**
Closes #389 

### DESCRIPTION ###

This is based on #402, which seems to be messed up somehow? (we're all scratching our heads) 

- [x] Drop sass-lint for stylelint
- [x] Update to Babel 7
- [x] Drop Sass Docs
- [x] I want linting to happen with each compile (see my example Webpack config, it does this)
- [x] BrowserSync stays!
- [x] NPM scripts all they way
- [x] Move browerlist to package.json
- [x] Update our linting configs to match WDS Code Standards
- [x] Added support for CSS Media Packer https://github.com/hail2u/node-css-mqpacker
- [x] NPM command added to build pot file
- [x] Support for spritesmith to generate image sprites added to webpack
- [x] minify images like gulp task
- [x] minify svgs like gulp task
- [x] generate svg icons sprite 
- [x] Added minification for JS on `npm build`

### OTHER ###
- [ ] Is this issue accessible? (Section 508/WCAG 2.0AA)
- [ ] Does this issue pass all the linting? (PHPCS, ESLint, SassLint)
- [ ] Does this pass CBT?

### STEPS TO VERIFY ###
How do we test this?

### DOCUMENTATION ###
Will this pull request require updating the wd_s [wiki](https://github.com/WebDevStudios/wd_s/wiki)?

### CODE REVIEW

**Comment by mharis** [comment]
- Worked on small improvements
- Spent time trying to figure out why browsersync keeps reloading. Maybe it needs to use webpack-dev-server.

**Comment by aaroneaton** [comment]
I got the eslint action fixed up. It was attempting to check a directory that didn't exist. Here's what the command is now in Buddy: `eslint './assets/scripts' --fix --debug --ignore-pattern lib/`

File: assets/scripts/index.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import 'normalize.css';
+import 'animate.css';
+import '../sass/style.scss';
+
+import './components/header-button.js';
```
Comment (coreymcollins) [inline assets/scripts/index.js:5]: Missing `accordion.js`. I had tried editing each of the files and the Accordion file would not compile on save.

File: webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+	plugins: [
+		new CleanPlugin( [ 'dist' ] ),
+		new SpritesmithPlugin(
+			{
+				src: {
+					cwd: path.resolve( __dirname, 'assets/images/sprites' ),
+					glob: '*.png'
+				},
+				target: {
+					image: path.resolve( __dirname, 'assets/images/sprites.png' ),
```
Comment (coreymcollins) [inline webpack.config.js:161]: This doesn't appear to be working. I've saved a couple of `.png` files in the `/assets/images/sprites/` directory and while it looks like webpack is doing something, the `assets/images/sprites.png` file is never updated.

**Review by coreymcollins — CHANGES_REQUESTED** [review]
I started testing this a bit today and have run into two items so far which I've noted with file comments. I've also asked Haris to put together a testing plan so we can test this out as robustly as possible, making sure we don't miss any updated features or anything like that.

So far, general Sass compiling is working well though I know we had talked about removing the need for a `rem()` mixin or limitations on what unit types could be used in our partials, instead, saving that for compiling. I thought that was going to be a part of this update as well, but maybe not? Currently, if I try to use something like a pixel value for padding instead of the `rem()` mixin, I'll get a compilation error.

Do we want to roll that in here or should that be reserved for another update?

File: assets/scripts/index.js (hunk 1 above)
Comment (mharis) [reply assets/scripts/index.js:5]: I had no such issue @coreymcollins. This is from compiled JS file. https://www.dropbox.com/s/cnytz9av853ipy9/Screenshot%202018-12-07%2017.35.12.png?dl=0

Each component module has to be imported within the entry point to be compiled by webpack. Our entry point for webpack is `assets/scripts/index.js`

File: webpack.config.js (hunk 2 above)
Comment (mharis) [reply webpack.config.js:161]: Yes, there was a small bug which took a while to fix but it is fixed now and pushed.

**Comment by mharis** [comment]
Passing this to @efuller for help. I am able to fix all the bugs I came across today except for this issue where I cannot make `url()` in sass files relative to the root of the theme folder. I'm not sure where I'm wrong and burnt out spending last 2 hours fixing this.

screenshot 2018-12-07 23 00 31

In the screenshot, URL is relative to the root of the website URL because it starts with a leading slash.

**Comment by efuller** [comment]
Did some investigating on the image issue and I have one working solution in place, but want to do some more exploring of options. 

I think the main thing I'm taking away here is that we may need to change up the URL we use when including images into our CSS. I also think we will need to make sure images are emitted over to the dist folder.

Getting close, but not quite there yet!

**Comment by coreymcollins** [comment]
@efuller & @mharis – do we still need this PR open with the existence of #452?

**Pull request closed without merging**